app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlAdapter.kt

```diff
@@ -305,7 +311,6 @@ class SessionControlAdapter(
             OnboardingToolbarPositionPickerViewHolder.LAYOUT_ID -> OnboardingToolbarPositionPickerViewHolder(
                 view
             )
-            MessageCardViewHolder.LAYOUT_ID -> MessageCardViewHolder(view, interactor)
             BottomSpacerViewHolder.LAYOUT_ID -> BottomSpacerViewHolder(view)
             else -> throw IllegalStateException()
         }
```

Comment:
This is missing an entry in `onViewRecycled` I think, but i would read up on https://github.com/mozilla-mobile/fenix/pull/24501#discussion_r866517396.